### DISCUSSION TITLE
Meta header for robots of noindex set by default preventing Google Index

### DISCUSSION THREAD

**Discussion by marduc812** [vercel/community#4852] [Help]
### Summary

I have a build on Vercel with next.js. For every page that it is not on `/` there is a the `<meta name="robots" content="noindex"/>` tag.

In `vercel.json `I set the HTTP headers to `{"key":"X-Robots-Tag", "value": "index, follow"}`.
I have a custom domain and it is set as production from Vercel's dashboard. 
I searched in my code and there is no meta tag for robots with a noindex value.

Am I missing something, or what should I do?

### Example

https://www.devoven.com/encoding/base64-encode
edit: Also link without cloudflare: https://dev-oven-v2.vercel.app/

### Steps to Reproduce

You can just see the source code. If there is something that could help, I will provide more info.

**Comment by marduc812** [comment]
Any suggestion or ideas what could cause that?
i ended up passing the meta tags like suggested and now it has both of them:

```
robots: {
    index: true,
    follow: true,
    nocache: true,
    googleBot: {
      index: true,
      follow: true,
      noimageindex: false,
      'max-video-preview': -1,
      'max-image-preview': 'large',
      'max-snippet': -1,
    }
  },
```

The HTML now has duplicate records like this:

```
<meta name="robots" content="noindex"/>
<meta name="robots" content="index, follow, nocache"/><meta name="googlebot" content="index, follow, max-video-preview:-1, max-image-preview:large, max-snippet:-1"/>
....
```

**Reply by z4nr34l** [reply]
Please try checking in incognito - I've check both URLs you provided and it look just fine:

For https://dev-oven-v2.vercel.app/
```html
<meta name="robots" content="index, follow, nocache">
<meta name="googlebot" content="index, follow, max-video-preview:-1, max-image-preview:large, max-snippet:-1">
```

For https://www.devoven.com/encoding/base64-encode
```html
<meta name="robots" content="index, follow, nocache">
<meta name="googlebot" content="index, follow, max-video-preview:-1, max-image-preview:large, max-snippet:-1">
```

I didn't found anything suspect there :D

**Reply by marduc812** [reply]
Hello @z4nr34l,

If you will see the source code of both URLs have the `noindex` in the robots header.

view-source:https://www.devoven.com/encoding/base64-encode
view-source:https://dev-oven-v2.vercel.app/encoding/base64-encode
 
 I noticed also that this header is present only in components that have client-site components and it is not present when I am running it in dev mode locally with `npm run dev`. Is it something `next` would add when compiling it, not sure.

**Comment by marduc812** [comment]
This behaviour is introduced from the use of `useSearchParams` from next.js. A workaround was suggested in the post below.

https://github.com/vercel/next.js/issues/58615#issuecomment-1820948414

**Reply by Dimirolz** [reply]
Thanks! That's helped
